src/data/matrix/notation.lean

```diff
@@ -249,16 +249,16 @@ lemma vec3_add [has_add α] (a₀ a₁ a₂ b₀ b₁ b₂ : α) :
   ![a₀, a₁, a₂] + ![b₀, b₁, b₂] = ![a₀ + b₀, a₁ + b₁, a₂ + b₂] :=
 by rw [cons_add_cons, cons_add_cons, cons_add_cons, empty_add_empty]
 
-variable [semiring α]
-
-lemma smul_vec2 (x : α) (a₀ a₁ : α) :
+lemma smul_vec2 [has_mul α] (x : α) (a₀ a₁ : α) :
   x • ![a₀, a₁] = ![x • a₀, x • a₁] :=
 by rw [smul_cons, smul_cons, smul_empty]
 
-lemma smul_vec3 (x : α) (a₀ a₁ a₂ : α) :
+lemma smul_vec3 [has_mul α] (x : α) (a₀ a₁ a₂ : α) :
   x • ![a₀, a₁, a₂] = ![x • a₀, x • a₁, x • a₂] :=
 by rw [smul_cons, smul_cons, smul_cons, smul_empty]
```

Comment:
Strictly the generalization is
```lean
lemma smul_vec3 [has_scalar R α] (x : R) (a₀ a₁ a₂ : α) :
  x • ![a₀, a₁, a₂] = ![x • a₀, x • a₁, x • a₂] :=
by rw [smul_cons, smul_cons, smul_cons, smul_empty]
```
Does this work for you?

Reply:
For the sake of proving the lemma, the following works.
```
lemma smul_vec2 {R : Type*} [has_scalar R α] (x : R) (a₀ a₁ : α) :
  x • ![a₀, a₁] = ![x • a₀, x • a₁] :=
by rw [smul_cons, smul_cons, smul_empty]

lemma smul_vec3 {R : Type*} [has_scalar R α] (x : R) (a₀ a₁ a₂ : α) :
  x • ![a₀, a₁, a₂] = ![x • a₀, x • a₁, x • a₂] :=
by rw [smul_cons, smul_cons, smul_cons, smul_empty]
```

However, I had to perform the same trick with `smul_vec3` in `cross_product` as we did with `vec3_add`. I will commit the change and you will tell me whether it is worth it.

Reply:
Have a look @eric-wieser and tell me whether it is better.
https://github.com/leanprover-community/mathlib/pull/11429/commits/9c302d38e4ae40f038baa56db45dd585e00db2e6
To me, the new version looks more complicated.

Reply:
I think we should keep it anyway - the complication is a good thing, as it makes it obvious that "something is wrong with lean", which will encourage future users to fix it.

Reply:
OK!